framework/rest/IndexAction.php

```diff
@@ -54,6 +78,10 @@ protected function prepareDataProvider()
             return call_user_func($this->prepareDataProvider, $this);
         }
 
+        if ($this->searchClass instanceof SearchInterface) {
+            return (new $this->searchClass)->search(Yii::$app->request->queryParams);
+        }
```

Comment:
To think: `IndexAction::prepareDataProvider` can be used for this.
Thus, it does not seem this entire PR make much sense.